Theano-cuda: init at 0.8.2

artuuge opened the pull request:
###### Motivation for this change

This is an extension of the existing python package `Theano-0.8.1` to add a functionality of running the code on an NVIDIA GPU. The minor upgrade to version 0.8.2 is required to provide compatibility with `libgpuarray--9998.0`. The `libgpuarray` package has a dependency on `clblas` which requires OpenCL 1.2. For this reason one needs to use `cudatoolkit-7.5` in place of `cudatoolkit-7.0`. 

The package builds fine with `python-3.5`, but many unit tests on GPU encounter a `Cannot allocate memory problem` error. The tests if one installs the package as `python34Pacakges.Theano-cuda` pass successfully:

``` bash
nix-shell -p pkgs.python34Packages.Theano-cuda
rm -rf ~/.theano
theano-test
```

Note that the `cuDNN` part of the tests requires a GPU of compute capability >= 3.0. 
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

artuuge committed:
Theano-cuda: init at 0.8.2

FRidh commented on pkgs/development/libraries/science/math/cudnn/7.5-5.0/default.nix:
```diff
@@ -0,0 +1,45 @@
+      This nix expression requires that ${name} is
+      already part of the store. Register yourself to NVIDIA Accelerated Computing Developer Program
+      and download cuDNN library at https://developer.nvidia.com/cudnn, and store it to the nix store with nix-store --add-fixed sha256 <FILE>.
+    '';
```
it is also possible to use `nix-prefetch-url` but I don't know whether one is preferred over the other.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -21029,6 +21029,31 @@ in modules // {
 
+  Theano-cuda = callPackage ../development/python-modules/theano/cuda (
+  let
+    numpy = self.numpy_1_10;
```
let's not use specific versions here

artuuge commented:
Let's use `nix-prefetch-url` to register the cuDNN library in the nix store. I have switched the versions of `numpy` and `scipy` to the default selections and squashed the commits.
